allennlp/interpret/influence_interpreters/simple_influence.py

```diff
@@ -0,0 +1,357 @@
+import json
+
+from typing import List, Optional, Tuple, Union
+import numpy as np
+from torch import Tensor
+from tqdm import tqdm
+import torch
+import torch.autograd as autograd
+from torch.nn import Parameter
+
+from allennlp.interpret.influence_interpreters.influence_interpreter import (
+    InfluenceInterpreter,
+)
+from allennlp.predictors import Predictor
+from allennlp.models.model import Model
+from allennlp.data import DatasetReader, Batch, Instance
+from allennlp.data.data_loaders import DataLoader, MultiProcessDataLoader
+from allennlp.nn import util
+
+
+@InfluenceInterpreter.register("simple-influence")
+class SimpleInfluence(InfluenceInterpreter):
+    """
+    Registered as a `SimpleInfluence` with name "simple-influence". This is a simple influence function
+    calculator. We simply go through every examples in train set to calculate the influence score, and uses
+    recommended LiSSA algorithm (essentially a first-order Talyor approxmation) to approximate the inverse
+    of Hessian used for influence score calculation. At best, we uses a single GPU for running the calculation.
+
+    # Parameter
+    predictor: `Predictor`
+        Required. This is a wrapper around the model to be tested. We only assume only `Model` is not None.
+    train_filepath: `str`
+        Required. This is the file path to the train data
+    test_filepath: `str`
+        Required. This is the file path to the test data
+    train_dataset_reader: `DatasetReader`
+        Required. This is the dataset reader to read the train set file
+    test_dataset_reader: `Optional[DatasetReader]` = None,
+        Optional. This is the dataset reader to read the test set file. If not provided, we would uses the
+        `train_dataset_reader`
+    params_to_freeze: Optional[List[str]] = None
+        Optional. This is a provided list of string that for freezeing the parameters. Expectedly, each string
+        is a substring within the paramter name you intend to freeze.
+    k: int = 20
+        Optional. To demonstrate each test data, we found it most informative to just provide `k` examples with the
+        highest and lowest influence score. If not provided, we set to 20.
+    device: int = -1,
+        Optional. The index of GPU device we want to calculate scores on. If not provided, we uses -1
+        which correspond to using CPU.
```

Comment:
Parameter descriptions for these parameters that the base class uses should go in the docstring for the base class, and then can be omitted here.

Reply:
Make sense!